catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/formulaeditor/FormulaEditorFunctionListTest.java

```diff
@@ -0,0 +1,180 @@
+				{R.string.formula_editor_function_max, R.string.formula_editor_function_max_parameter, "max"},
+				{R.string.formula_editor_function_min, R.string.formula_editor_function_min_parameter, "min"},
+				{R.string.formula_editor_function_length, R.string.formula_editor_function_length_parameter, "length"},
+				{R.string.formula_editor_function_letter, R.string.formula_editor_function_letter_parameter,
+						"letter"},
+				{R.string.formula_editor_function_join, R.string.formula_editor_function_join_parameter, "join"},
+				{R.string.formula_editor_function_number_of_items, R.string
+						.formula_editor_function_number_of_items_parameter, "number of items"},
+				{R.string.formula_editor_function_list_item, R.string.formula_editor_function_list_item_parameter,
+						"list item"},
+				{R.string.formula_editor_function_contains, R.string.formula_editor_function_contains_parameter,
+						"contains"}
+		});
+	}
+
+	@Parameter
+	public @StringRes int formulaEditorFunction;
+
+	@Parameter(1)
+	public @StringRes int formulaEditorFunctionParameter;
+
+	@Parameter(2)
+	public String testName;
+
+	@Before
+	public void setUp() throws Exception {
+		Script script = BrickTestUtils.createProjectAndGetStartScript("FormulaEditorFunctionListTest");
+		script.addBrick(new ChangeSizeByNBrick(0));
+		baseActivityTestRule.launchActivity(null);
+	}
+
+	@Test
+	public void testFunctionsListElements() {
+		onBrickAtPosition(0).checkShowsText(R.string.brick_when_started);
+		onBrickAtPosition(1).checkShowsText(R.string.brick_change_size_by);
+		onBrickAtPosition(1).perform(click());
+
+		onView(withText(R.string.formula_editor_functions)).perform(click());
+
+		String formulaEditorFunctionString = InstrumentationRegistry.getTargetContext()
+				.getString(formulaEditorFunction);
+
+		String formulaEditorFunctionParameterString = InstrumentationRegistry.getTargetContext()
+				.getString(formulaEditorFunctionParameter);
+
+		String editorFunction = formulaEditorFunctionString + formulaEditorFunctionParameterString;
+
+		String selectedFunctionString = getSelectedFunctionString(formulaEditorFunctionString,
+				formulaEditorFunctionParameterString);
+
+		onData(allOf(is(instanceOf(String.class)), is(editorFunction)))
+				.inAdapterView(FormulaEditorFunctionListMatchers.isFunctionListView())
+				.check(matches(isDisplayed()));
+
+		onView(withText(editorFunction)).perform(click());
+
+		onView(withId(R.id.formula_editor_edit_field)).check(matches(withText(selectedFunctionString)));
+	}
+
+	private String getSelectedFunctionString(String function, String functionParameter) {
+		String selectedFunctionString = function;
+		Boolean bracketsUsed = functionParameter.contains("(") | functionParameter.contains(")");
+
+		if (bracketsUsed) {
+			selectedFunctionString += "(";
+			functionParameter = functionParameter.substring(1, functionParameter.length() - 1);
+		}
+
+		String[] parameterArray = functionParameter.split(",");
```

Comment:
please make this an iterable and then use a for/each to iterate...

Reply:
ok, after reading this to the end, theres no loop or anything necessary.
all parameter strings start with "(" and end with ")", besides no_parameter ones.
the spacing can be easily done by someString.replace(","," , ")